### DISCUSSION TITLE
Update trained model by training on new videos

### DISCUSSION THREAD

**Discussion by Advika221** [talmolab/sleap#1601] [Help!] [answered]
Hi, I have a single instance model trained using one video. I found it is not predicting the rodent's position accurately in other videos with other objects in the frame. I am trying to update this model by using the **_same_** model and training it using random frames from my other videos. However, I don't think sleap is updating my original model. It is making new models instead. 

(1) How can I update my original model by training on new videos?
(2) Are the new models made from using my old model to train on new videos a combination of my old model + new model created?

Thanks for any advice!

**Comment by talmo** [comment, marked as answer]
Hi @Advika221,

Thanks for reaching out. In order to train a SLEAP model on both the old and new data, you'll want to make sure that both videos are included in the **same SLEAP labels project**.

You can do this by adding them both in the **Videos** tab.

If you've already started labeling on two separate projects, you can merge them into a combined project by going to **File** --> **Merge Into Project...** and select the other SLP file to merge into your current project.

Afterwards, the combined set of labels from both should appear and any new models trained will include the labels from both (and therefore should work on both kinds of videos).

As an additional clarification:

When you have other models in the `models/` subfolder in the same directory as your SLP file, they'll appear in the dropdown in the **Run Training** window, with a "**[Trained]**" prefix next to each model’s training_config.json.

You can use this feature to quickly reuse the settings you had configured previously, but you'll be training a completely new model.

When you select "**Use Trained Model**", all the fields will be grayed out no training will occur for that model and SLEAP will just use the previously trained model.

If you select "**Resume Training**", however, what SLEAP does is to train a new model, but it initializes it with the previous model weights. This is what is known as "transfer learning", or essentially reusing part of what it learned in the previous run as a starting point. This can help when you have a model trained on a different but related dataset that can bootstrap your model optimization. It does _not_, however, have any guarantee that it'll "remember" the old data nor work very well on that data. To train a model that works well on both datasets, you have to include both datasets in the same SLEAP project.

Let us know if you have any questions!

Talmo

**Reply by Advika221** [reply]
Perfect. Thanks so much!
